ciao-controller/internal/datastore/sqlite3db.go

```diff
@@ -2304,9 +2307,26 @@ func (ds *sqliteDB) deleteBlockData(ID string) error {
 }
 
 func (ds *sqliteDB) createStorageAttachment(a types.StorageAttachment) error {
-	ds.dbLock.Lock()
-	err := ds.create("attachments", a.ID, a.InstanceID, a.BlockID)
-	ds.dbLock.Unlock()
+	datastore := ds.getTableDB("attachments")
+
+	ds.tdbLock.Lock()
+
```

Comment:
For the "attachments" table, you need to use dbLock, not tdbLock. It would have been nice if I'd designed "getTableDB" to also return the lock you need to use, but alas.